### PR TITLE
feat: use aliyun oss sdk to visit oss

### PR DESCRIPTION
**Pull request by sysvm**
### Description

PieceStore uses aliyun OSS sdk to visit OSS.

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 
* N/A

### CODE REVIEW

**Review by will-2012 — APPROVED** [review]

**Review by flywukong — APPROVED** [review]

File: store/piecestore/storage/b2.go [hunk 1]
```diff
@@ -98,6 +98,6 @@ func parseB2BucketURL(bucketURL string) (endpoint, region, bucketName string, er
 		isVirtualHostStyle = true
 	}
 
-	region = parseRegion(endpoint)
+	region = parseS3Region(endpoint)
```
Comment (RenRick) [inline store/piecestore/storage/b2.go:101]: Why need to change B2?

File: store/piecestore/storage/oss.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+func (o *ossStore) String() string {
+	return fmt.Sprintf("oss://%s/", o.bucket.BucketName)
+}
+
+func (o *ossStore) CreateBucket(ctx context.Context) error {
+	err := o.bucket.Client.CreateBucket(o.bucket.BucketName)
+	if err != nil && isErrExists(err) {
+		err = nil
+	}
+	return err
```
Comment (RenRick) [inline store/piecestore/storage/oss.go:34]: Don't need to return bucket besides error?

File: store/piecestore/storage/oss.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	}
+	return resp, err
+}
+
+func (o *ossStore) PutObject(ctx context.Context, key string, in io.Reader) error {
+	var option []oss.Option
+	if ins, ok := in.(io.ReadSeeker); ok {
+		option = append(option, oss.Meta(ChecksumAlgo, generateChecksum(ins)))
+	}
+	var respHeader http.Header
```
Comment (RenRick) [inline store/piecestore/storage/oss.go:62]: Better to group var

File: store/piecestore/storage/oss.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	option = append(option, oss.GetResponseHeader(&respHeader))
+	err := o.bucket.PutObject(key, in, option...)
+	return err
+}
+
+func (o *ossStore) DeleteObject(ctx context.Context, key string) error {
+	var respHeader http.Header
+	err := o.bucket.DeleteObject(key, oss.GetResponseHeader(&respHeader))
```
Comment (RenRick) [inline store/piecestore/storage/oss.go:70]: return o.bucket.... directly

File: store/piecestore/storage/b2.go (hunk 1 above)
Comment (sysvm) [reply store/piecestore/storage/b2.go:101]: Because B2 uses s3 `parseRegion` funciton

File: store/piecestore/storage/oss.go (hunk 3 above)
Comment (sysvm) [reply store/piecestore/storage/oss.go:62]: I'll fix it in another pr.

File: store/piecestore/storage/oss.go (hunk 4 above)
Comment (sysvm) [reply store/piecestore/storage/oss.go:70]: I'll fix it in another pr.

File: store/piecestore/storage/oss.go (hunk 2 above)
Comment (sysvm) [reply store/piecestore/storage/oss.go:34]: If users create bucket successfully, they can do some operations based on bucketName.

**Pull request merged**